datafusion/expr/src/udf.rs

```diff
@@ -292,3 +271,105 @@ pub trait ScalarUDFImpl {
         &[]
     }
 }
+
+/// ScalarUDF that adds an alias to the underlying function. It is better to
```

Comment:
This is somewhat boilerplate, but it is a pretty straightforward example of using Trait objects to extend functionality